src/index.ts

```diff
@@ -22,6 +22,28 @@ export const configs = {
       ],
     },
   },
+  next: {
+    name: "react-refresh/vite",
+    plugins: { "react-refresh": plugin },
+    rules: {
+      "react-refresh/only-export-components": [
+        "error",
+        {
+          // from https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config
+          allowExportNames: [
```

Comment:
@ArnaudBarre I just discovered that this is also allowed:

`viewport`

See https://nextjs.org/docs/app/api-reference/functions/generate-viewport.